platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -270,6 +272,9 @@ export function ViewportGridProvider({ children, service }) {
               viewports
             );
           }
+          viewport.viewportOptions.viewportId ||= `viewport-${viewportIndex}`;
```

Comment:
Similar question as before, why the ||?

Reply:
Because the provided viewport options may not have an id provided already.

Reply:
Sure. But from what I have witnessed, I have NOT seen any viewport with an id of say 'viewport-2'. I HAVE seen viewports with ids like '0-0', '0-1', '1-1', etc. So I am still confused. Did I see wrong?

Reply:
If you run a code path that calls setDisplaySetsForViewports, and provides viewportOptions, but not viewportId, then you would see it - try displaying an RTStruct using Alireza's new code.  That one doesn't include the viewportId, so you will see the viewport-2 type of id.
Not saying it is common, just that I don't want it forgotten.

Reply:
Ok. Well I MIGHT try to rebase your branch on the RTStruct to see what you mean, but only if there are no conflicts. However, does this mean there are potentially THREE different UIDs so to speak...
- a custom one in the HP/stage
- this `viewport-{index}` thingy
- grid positioning (i.e. `0-1`)

Reply:
Yes, there might be different ones coming from different origins.  The format of it doesn't really matter, as long as they are unique at the time of render